docs: improve guides (#799)
biomejs/website: src/content/docs/guides/how-biome-works.mdx

The configuration file is considered **optional**, Biome has good defaults. Use the configuration
file to change those defaults.

The Biome configuration file is named `biome.json` or `biome.jsonc` and should be placed in the root directory of your project.
The root directory is usually the directory containing your project's `package.json`.

:::note
If both `biome.json` and `biome.jsonc` are present in the same folder, the priority will be given to `biome.json`
:::

> The following configuration file enables the formatter, sets the indent style and line width, and disabled the linter:

Reviewer: ```suggestion
The following configuration file enables the formatter, sets the indent style and line width, and disables the linter:
```

will use the configuration file `app/frontend/biome.json`;

:::note

> The Biome commands support using the `--config-path` option. This allows you to specify a custom configuration file or a directory where Biome can look for a `biome.json` or `biome.jsonc` file. When you use `--config-path`, the standard configuration file resolution process is **disabled**.

Reviewer: ```suggestion
Biome commands support using the `--config-path` option. This allows you to specify a custom configuration file or a directory where Biome can look for a `biome.json` or `biome.jsonc` file. When you use `--config-path`, the standard configuration file resolution process is **disabled**.
```

}
```

When you run the script `format` from `frontend/package.json`,
the working directory resolved by that script will be `frontend/`.
The globs `src/**/*.js` and `src/**/*.ts` will have as "base" directory `frontend/`.

At the moment, Biome uses a glob library that treats all globs as having a `**/` prefix.
This means that `src/**/*.js` and `**/src/**/*.js` are treated as identical.
They match both `src/file.js` and `test/src/file.js`.

> This is something we plan to fix in Biome 2, as it will be a breaking change.

Reviewer: ```suggestion
This is something we plan to fix in Biome **v2.0.0**, as it will be a breaking change.
```
